conda_env/cli/main_create.py

```diff
@@ -51,22 +53,36 @@ def configure_parser(sub_parsers):
         '-q', '--quiet',
         default=False,
     )
+    p.add_argument(
+        'handle',
```

Comment:
I don't think issue 1 is an issue.  When you run `hub clone tswicegood/conda` you get `conda` as a sub-directory as the portion before the `/` is part of the namespace.  Same with `git clone git://github.com/tswicegood/conda.git`.  The last element in the URI provides the leaf node, which is what's assumed to be the name as that's where the tree terminates.

I'd prefer to do the name deprecation as part of this so we have it all in one take instead of multiple releases.  This is driven by wanting to do `conda env create web-dev gulp` and have that translate to `conda create -n web-dev gulp`.  I can see another case where `conda env create tswicegood/web-dev broccoli` would create an environment from `tswicegood/web-dev` but also add `broccoli` to it.

With the deprecation, I see `conda env create tswicegood/web-dev` and `conda env create --name tswicegood/web-dev` being equivalent with the latter displaying a deprecation warning.